### PR TITLE
Fix p2p header adapters

### PR DESCRIPTION
**Pull request by kirugan**

### CODE REVIEW

File: adapters/core2p2p/block.go [hunk 1]
```diff
@@ -45,16 +45,15 @@ func AdaptHeader(header *core.Header, commitments *core.BlockCommitments,
...
-		Receipts:        AdaptHash(&felt.Zero),
+		Receipts:        AdaptHash(commitments.ReceiptCommitment),
 		ProtocolVersion: header.ProtocolVersion,
-		GasPriceFri:     AdaptUint128(header.GasPrice),
+		GasPriceFri:     AdaptUint128(header.GasPriceSTRK),
 		Signatures:      utils.Map(header.Signatures, AdaptSignature),
 		StateDiffCommitment: &spec.StateDiffCommitment{
 			StateDiffLength: stateDiffLength,
-			Root:            AdaptHash(stateDiffCommitment),
+			Root:            AdaptHash(stateDiffCommitment), // todo commitments
```
Comment (kirugan) [inline adapters/core2p2p/block.go:54]: I want to debug why it works for pathfinder if we pass `commitments.StateDiffCommitment` here

Comment (kirugan) [reply adapters/core2p2p/block.go]: I passed incorrect value initially it should be not `stateDiff.Commitment()`, but `stateDiff.Hash()`

**Comment by rianhughes** [comment]
`func (d *StateDiff) Commitment() *felt.Felt {...}`

doesn't seem to be used anywhere, can we remove it?

**Review by rianhughes — APPROVED** [review]

**Comment by kirugan** [comment]
> doesn't seem to be used anywhere, can we remove it?

I think yes, but I'm checking with Aneeque. Will remove in next PR if we don't need it anymore

**Review by AnkushinDaniil — APPROVED** [review]

**Pull request merged**